cmd/controller-manager/app/controllermanager.go

```diff
@@ -741,6 +741,10 @@ func setupControllers(mgr controllerruntime.Manager, opts *options.Options, stop
 	sharedFactory.Start(stopChan)
 	sharedFactory.WaitForCacheSync(stopChan)
 
+	if err := helper.IndexWork(context.TODO(), mgr); err != nil {
+		klog.Fatalf("Failed to index Work: %v", err)
+	}
```

Comment:
I think we can build the indexer at the place where configure the controller manager because it's kind of an overall configuration.
`setupControllers` is mostly for configure controllers.
 https://github.com/karmada-io/karmada/blob/20af47da0b968bd83cae482c0017f1346071a036/cmd/controller-manager/app/controllermanager.go#L193

We can set the index just before setupControllers.

Reply:
updated